projects/models.nix

```diff
@@ -0,0 +1,128 @@
+{
+  lib,
+  pkgs,
+  sources,
+}:
+let
+  yants = import sources.yants { };
+
+  inherit (yants)
+    string
+    list
+    option
+    attrs
+    either
+    struct
+    drv
+    path
+    restrict
+    defun
+    function
+    any
+    ;
+
+  programType = struct "program" {
+    name = option string;
+    module = defun [
+      (attrs any)
+      (attrs any)
+    ] function any;
```

Comment:
I'm not sure on the exact syntax here, can you please add an example to check?

Reply:
I'd like to see the shape of an actual module, namely `{...}: {}`. Maybe I read the docs too superficially, but I understood we'd write `defun [(attrs any) (attrs any)]` and then apply `{...}: {}` as a value to be type-checked.

Reply:
There were issues making `defun` work inside structs, but when it's it's defined outside of it, then it works as you've described. I've included a slightly more elaborate example than a `{...}: {}` to show the result.

Reply:
Yeah sorry I actually did misread the docs. `defun` does runtime input/output typechecks, and while that's nice we don't need or want that here. See the other comment